arbor/backends/gpu/stack.hpp

```diff
@@ -67,41 +81,67 @@ class stack {
         storage_(create_storage(capacity)), gpu_context_(gpu_ctx) {}
 
     ~stack() {
-        storage_.synchronize();
-        if (storage_->data) {
-            allocator<value_type>().deallocate(storage_->data, storage_->capacity);
+        memory::cuda_sync();
+        auto st = get_storage_copy();
```

Comment:
We don't need to update the host copy of the data to do the deallocation (we have the pointer already on the host side).

Reply:
Again, you don't need to update the host copy.